includes/hub/class-nodes.php

```diff
@@ -28,6 +27,7 @@ class Nodes {
 	public static function init() {
 		add_action( 'init', [ __CLASS__, 'register_post_type' ] );
 		add_action( 'save_post', [ __CLASS__, 'save_post' ] );
+		add_action( 'save_post_' . self::POST_TYPE_SLUG, [ __CLASS__, 'sync_nodes' ] );
```

Comment:
Pleas use the `newspack_network_node_saved` action. We have had race condition issues around this https://github.com/Automattic/newspack-network/pull/57

unrelated to this PR, but maybe as a safeguard we should make the callback that saves the nodes run on an higher priority (less then 10) here:

https://github.com/Automattic/newspack-network/blob/a9230d9e20cb9f59c78b56e6125d047b4be1386d/includes/hub/class-nodes.php#L30

Reply:
Sure! One issue with using this action is when a node is trashed from the edit posts page vs. the edit single post page. In this case the nonce is not present and so the sync event won't trigger. 

Is it possible to inject the nonce field to the edit posts page? If not, I guess we can either replace the trash and restore URLs to include our nonces, or simply remove the option to trash from the edit posts page altogether.

I'd love your thoughts on this before I make any changes here.

Reply:
I think you could listen to an additional hook to trigger this then

for example you listen to `newspack_network_node_saved` and also to `trashed_post`, calling the same callback.

Reply:
> for example you listen to newspack_network_node_saved and also to trashed_post, calling the same callback.

This works too! I wasn't sure if we wanted to avoid using the standard wordpress post hooks to avoid running into similar issues in the future. In any case I made the change in https://github.com/Automattic/newspack-network/pull/56/commits/1e590d1458290550fb8e45cc40e445bbeffca0a6.

I also added some np network specific hooks for trashing and untrashing nodes in case we need to do more than just sync nodes in the future. This might be overkill though, so if you'd like I can remove these and just hook directly on the `un/trashed_post` hooks.